### PR TITLE
Document `table_changes` Iceberg table function

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Documentation follow-up from https://github.com/trinodb/trino/pull/15677

## Additional context and related issues

Fixes https://github.com/trinodb/trino/issues/21227

Iceberg Spark counterpart https://iceberg.apache.org/docs/latest/spark-procedures/#create_changelog_view

Delta Lake `table_changes` table function documenation https://trino.io/docs/current/connector/delta-lake.html#table-changes

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.md
Comment (findinpath) [inline docs/src/main/sphinx/connector/iceberg.md:1544]: `SCHEMA` and `TABLE` are reserved words, so we'll have to quote them to avoid unwanted sql syntax issues.

cc @martint

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:1544]: Thats kinda bad .. why did those parameters not use the same approach as in https://trino.io/docs/current/connector/iceberg.html#migrate-table

so schema_name
and table_name

and also stick with lower case parameter names

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:1544]: could we maybe fix the implementation ?

Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:1544]: https://github.com/trinodb/trino/pull/21698 fixes the issue. 

Thank you @ebyhr 

Reviving this PR

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1534]:
```suggestion
The connector supports the table functions described in the following sections.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1539]:
```suggestion
The following query shows an example of displaying the changes of the `t1` table in the `default` schema in the current catalog. All changes between the start and end snapshots are returned.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1555]: I think we can leave the type out for all these parameters. Also maybe use the same list layout as in the following section.

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1559]: maybe we should link to a section how to find out the snapshot ids

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1563]:
```suggestion
The function returns the columns present in the table, and the following values for each  change request:
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1564]: remove

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1567]:
```suggestion
  : The type of change that occurred. Possible values are `insert` and `delete`.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1569]:
```suggestion
  : The identifier of the snapshot in which the change occurred.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1571]:
```suggestion
  : Timestamp for the moment when the snapshot became active.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1573]:
```suggestion
  : Order number of the change, useful for sorting the results.
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1575]: Just add

**Example**

instead

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1577]:
```suggestion
Create a table:
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1583]:
```suggestion
Insert some data:
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1598]:
```suggestion
Retrieve the snapshot identifiers of the changes:
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1618]:
```suggestion
Select the changes performed in the previously-mentioned `INSERT` statements:
```

File: docs/src/main/sphinx/connector/iceberg.md
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:1590]: are the domain values here and below different on purpose?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/iceberg.md
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:1590]: the `page_url` & `domain` are supposed to be the primary key so to say of the table that can be used in downstream processes to identify the history of changes performed on a tuple.

Note that the same example slightly adapted is employed in Trino Delta Lake connector as well.

**Review by mosabua — APPROVED** [review]
Looks good now.  Ideally add the sql language identifier to all the query code snippets .. also not sure if you want approval from another maintainer.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**
